pypesto/objective/priors.py

```diff
@@ -0,0 +1,264 @@
+
+        return hessian
+
+    def hessian_vp_for_full_parameter_vector(self, x, p):
+
+        h_dot_p = np.zeros_like(p)
+
+        for prior in self.prior_list:
+            h_dot_p[prior['index']] -= \
+                prior['density_ddx'](x[prior['index']]) * p[prior['index']]
+
+        return h_dot_p
+
+
+def get_parameter_prior_dict(index: int,
+                             prior_type: str,
+                             prior_parameters: list,
+                             parameter_scale: str = 'lin'):
+
+    """
+    Returns the prior dict used to define priors for some default priors.
+
+    index:
+        index of the parameter in x_full
+
+    prior_type: str
+        Prior is defined in LINEAR parameter space! prior_type can from
+        {uniform, normal, laplace, logUniform, logNormal, logLaplace}
+
+    prior_parameters:
+        Parameters of the priors. Parameters are defined in linear scale.
+
+    parameter_scale:
+        scale, in which parameter is defined (since a parameter can be
+        log-transformed, while the prior is always defined in the linear space)
+    """
+
+    log_f, d_log_f_dx, dd_log_f_ddx = \
+        _prior_densities(prior_type, prior_parameters)
+
+    if parameter_scale == 'lin':
+
+        return {'index': index,
+                'density_fun': log_f,
+                'density_dx': d_log_f_dx,
+                'density_ddx': dd_log_f_ddx}
+
+    elif parameter_scale == 'log':
+
+        def log_f_log(x_log):
+            """log-prior for log-parameters"""
+            return log_f(math.exp(x_log))
+
+        def d_log_f_log(x_log):
+            """derivative of log-prior w.r.t. log-parameters"""
+            return d_log_f_dx(math.exp(x_log)) * math.exp(x_log)
+
+        def dd_log_f_log(x_log):
+            """second derivative of log-prior w.r.t. log-parameters"""
+            return math.exp(x_log) * \
+                (d_log_f_dx(math.exp(x_log)) +
+                    math.exp(x_log) * dd_log_f_ddx(math.exp(x_log)))
+
+        return {'index': index,
+                'density_fun': log_f_log,
+                'density_dx': d_log_f_log,
+                'density_ddx': dd_log_f_log}
+
+    elif parameter_scale == 'log10':
+
+        log10 = math.log(10)
+
+        def log_f_log10(x_log10):
+            """log-prior for log10-parameters"""
+            return log_f(10**x_log10)
+
+        def d_log_f_log10(x_log10):
+            """derivative of log-prior w.r.t. log10-parameters"""
+            return d_log_f_dx(10**x_log10) * log10 * 10**x_log10
+
+        def dd_log_f_log10(x_log10):
+            """second derivative of log-prior w.r.t. log10-parameters"""
+            return log10**2 * 10**x_log10 * \
+                   (dd_log_f_ddx(10**x_log10) * 10**x_log10
+                    + d_log_f_dx(10**x_log10))
+
+        return {'index': index,
+                'density_fun': log_f_log10,
+                'density_dx': d_log_f_log10,
+                'density_ddx': dd_log_f_log10}
+
+    else:
```

Comment:
formulas look all right, but to ensure no errors come in when making changes, all should be covered by finite different tests.